### PR TITLE
statistics: do not analyze non exist table anymore

### PR DESCRIPTION
**Pull request by Rustin170506**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/57227

Problem Summary:

### What changed and how does it work?

After https://github.com/pingcap/tidb/pull/57222, we can handle most cases where tables have been dropped. However, there is still a chance that we might attempt to analyze a non-existent table or use incorrect information to perform the analysis.

In this PR, I check the schema and table information before performing the actual analysis. This helps minimize unnecessary analysis work.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/statistics/handle/autoanalyze/priorityqueue/dynamic_partitioned_table_analysis_job.go [hunk 1]
```diff
@@ -35,50 +36,54 @@ const (
...
 	// Some indicators to help us decide whether we need to analyze this table.
 	Indicators
-	GlobalTableID int64
-
 	// Analyze table with this version of statistics.
 	TableStatsVer int
 	// Weight is used to calculate the priority of the job.
 	Weight float64
+
+	// Lazy initialized.
```
Comment (Rustin170506) [inline pkg/statistics/handle/autoanalyze/priorityqueue/dynamic_partitioned_table_analysis_job.go:57]: Maybe we should put these fields into a substruct. But the PR is too big. I will do it later.

File: pkg/statistics/handle/autoanalyze/priorityqueue/non_partitioned_table_analysis_job.go [hunk 2]
```diff
@@ -34,34 +35,39 @@ const (
...
+	SchemaName pmodel.CIStr
+	TableID    int64
+	IndexIDs   map[int64]struct{}
+
 	Indicators
-	TableID       int64
 	TableStatsVer int
 	Weight        float64
+
+	// Lazy initialized.
```
Comment (Rustin170506) [inline pkg/statistics/handle/autoanalyze/priorityqueue/non_partitioned_table_analysis_job.go:52]: ditto

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Review by time-and-fate — COMMENTED** [review]
Others LGTM.

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**